### PR TITLE
Add fully dynamic nav component

### PR DESCRIPTION
**Pull request by WryanCartie**
Implemented a fully dynamic nav component to deal with links and general navigation.

### CODE REVIEW

File: src/shared/Navigation/Nav.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<template>
+    <b-nav id="nav" :vertical="vertical || false" :class="styling" :align="align || ''">
+        <b-nav-item v-for="item in items" :active="item.active" v-bind:key="item.text">
```
Comment (extremebip) [inline src/shared/Navigation/Nav.vue:3]: Add onclick event here

File: src/shared/Navigation/Nav.vue [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<template>
+    <b-nav id="nav" :vertical="vertical || false" :class="styling" :align="align || ''">
```
Comment (extremebip) [inline src/shared/Navigation/Nav.vue:2]: Just in case there will be multiple navbars in one page, make the `id` dynamic with `props`

Comment (WryanCartie) [reply src/shared/Navigation/Nav.vue:2]: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Review by extremebip — CHANGES_REQUESTED** [review]
Remove extra spaces from line 2 to 22

**Comment by Phobez** [comment]
@WryanCartie @extremebip don't forget :v

**Pull request closed without merging**